engine/src/main/java/org/destinationsol/removal/systems/DestructionSystem.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2020 The Terasology Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.destinationsol.removal.systems;
+
+import org.destinationsol.entitysystem.EventReceiver;
+import org.destinationsol.removal.components.SlatedForDeletion;
+import org.destinationsol.removal.events.DeletionEvent;
+import org.destinationsol.removal.events.ShouldBeDestroyedEvent;
+import org.destinationsol.removal.events.RemovalForOptimizationEvent;
+import org.terasology.gestalt.entitysystem.entity.EntityRef;
+import org.terasology.gestalt.entitysystem.event.EventResult;
+import org.terasology.gestalt.entitysystem.event.ReceiveEvent;
+
+import java.util.HashSet;
+
+/**
+ * When an entity receives a {@link ShouldBeDestroyedEvent} or a {@link RemovalForOptimizationEvent}, this system adds a
+ * {@link SlatedForDeletion} component. Any system that should respond to the destruction of an entity before it happens
```

Comment:
Does this at all get into the priority levels available on events? That's another thing I'm used to from TS (make events _high priority_ if they're meant to get processed before other key events) - not totally sure if it is a thing in latest Gestalt and/or DS